### DISCUSSION TITLE
Status: 500. Message: db query error

### DISCUSSION THREAD

**Discussion by gisxlab** [teslamate-org/teslamate#3640] [Q&A] [answered]
Please help me: my deployment seems to be successful, and I deployed successfully once on the previous server, then I deleted and redeployed and it never succeeded again. I tried different environments, CentOS, Debian and some Linux panels with no success, either a simple deployment via Docker or an Advanced installation with Traefik, Let's Encrypt & HTTP Basic Auth.
Currently, my project can run successfully, and both the TeslaMate page and Grafana page can be opened successfully, but the data cannot be displayed.

Status: 500. Message: db query error: pq: syntax error at or near ";"
Status: 500. Message: db query error: pq: syntax error at or near "order"
Status: 500. Message: db query error: pq: invalid input syntax for type json

![image](https://github.com/teslamate-org/teslamate/assets/35869860/aaa8620d-9b05-4482-9736-7bf68b839cb1)
![image](https://github.com/teslamate-org/teslamate/assets/35869860/55d89386-5bf4-4588-986f-28b1445d6571)

`version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    environment:
      - ENCRYPTION_KEY=LCaMFGC8obWRHKMtkN8wUMxErYy6K
      - DATABASE_USER=teslamate
      - DATABASE_PASS=ufHi8oyjHBTjWCEYehnHDsJXymEW8
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
      - TZ=Asia/Shanghai
    ports:
      - 4000:4000
    volumes:
      - ./import:/opt/app/import
    cap_drop:
      - all

  database:
    image: postgres:15
    restart: always
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD=ufHi8oyjHBTjWCEYehnHDsJXymEW8
      - POSTGRES_DB=teslamate
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=ufHi8oyjHBTjWCEYehnHDsJXymEW8
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3000:3000
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    # ports:
    #   - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:`

**Comment by banujan59** [comment]
I have the same issue. If I try to run the following SQL command on my DB:
SELECT * FROM  cars LIMIT  50
0 results are returned. 

The car was not added to the database (even though I successfully logged in using my tokens). 

This causes your other queries to be messed up. For example, in the charge detail dashboard the query is missing the "p.car_id" value near the "ORDER BY" syntax

![Screenshot from 2024-01-30 21-49-32](https://github.com/teslamate-org/teslamate/assets/16767628/47d3630c-5d23-4a41-b768-5002c2d397a3)

**Reply by wgspringfree** [reply]
Same issue here. Login to teslamate with the correct tokens. Got the same error. How to resolve this issue?

**Comment by socekin** [comment]
Same issue here, anyone solved this ?

**Comment by MooreOfNick** [comment]
I know it's been a while, but I'm also having this same issue now as well.

**Comment by mlaventure** [comment, marked as answer]
Please make sure you've updated to the latest docker-compose.yml, it's using a newer version of postgres: https://docs.teslamate.org/docs/installation/docker

**Reply by mlaventure** [reply]
Specifically, you'll need to migrate the data too, the instructions are in the release notes. For the latest minor 1.31 release check them out over there: https://github.com/teslamate-org/teslamate/releases/tag/v1.31.0

**Reply by SamuraiKK** [reply]
@mlaventure Even after updating to the latest docker-compose, I'm still getting same 500 errors. Please advice if there is a fix for it.

**Reply by mlaventure** [reply]
Did you migrate your data as per the release notes? That was the missing piece for me.
